### PR TITLE
[core] Introduce timeout for commit retry avoid long time loop

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose
1. When execute drop partition or expire partition with large numbers partitions(eg: table with large number of files and stream job with 1 min ckp), would block batch job for overwrite in a long time loop,  so add a timeout option for commit retry, user can quick get job result in a specific timeout to handle their batch job(which can adjust ckp time interval or stop stream job to finish their batch job). 
2. FileLength change to 4000 for CoreOption lines was near 3000 which would block code compile.

Linked issue: close #xxx

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 1]
```diff
@@ -134,6 +135,7 @@ public class FileStoreCommitImpl implements FileStoreCommit {
     private final List<CommitCallback> commitCallbacks;
     private final StatsFileHandler statsFileHandler;
     private final BucketMode bucketMode;
+    private final Optional<Duration> commitMaxTimeout;
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:138]: do not use optional for class member.

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 2]
```diff
@@ -1066,6 +1082,16 @@ public void compactManifest() {
                                 commitMaxRetries));
             }
             retryCount++;
+
+            if (commitMaxTimeout.isPresent()
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:1086]: Please merge into `if (retryCount >= commitMaxRetries)`.

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 3]
```diff
@@ -754,6 +759,16 @@ private int tryCommit(
...
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:763]: Please merge into if (retryCount >= commitMaxRetries).

File: docs/layouts/shortcodes/generated/core_configuration.html [hunk 4]
```diff
@@ -134,6 +134,12 @@
             <td>Integer</td>
             <td>Maximum number of retries when commit failed.</td>
         </tr>
+        <tr>
+            <td><h5>commit.max-timeout</h5></td>
```
Comment (JingsongLi) [inline docs/layouts/shortcodes/generated/core_configuration.html:138]: max timeout seems weird... it is just timeout?
`commit.timeout`

Comment (xuzifu666) [reply docs/layouts/shortcodes/generated/core_configuration.html]: Make sense，changed to commit.timeout，Thanks~ @JingsongLi

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 5]
```diff
@@ -166,7 +168,8 @@ public FileStoreCommitImpl(
             BucketMode bucketMode,
             @Nullable Integer manifestReadParallelism,
             List<CommitCallback> commitCallbacks,
-            int commitMaxRetries) {
+            int commitMaxRetries,
+            @Nullable Duration commitMaxTimeout) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:172]: Can you modify all MaxTimeout to Timeout?

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 6]
```diff
@@ -134,6 +135,7 @@ public class FileStoreCommitImpl implements FileStoreCommit {
     private final List<CommitCallback> commitCallbacks;
     private final StatsFileHandler statsFileHandler;
     private final BucketMode bucketMode;
+    @Nullable private Duration commitMaxTimeout;
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:138]: store millis here.

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 7]
```diff
@@ -134,6 +134,7 @@ public class FileStoreCommitImpl implements FileStoreCommit {
     private final List<CommitCallback> commitCallbacks;
     private final StatsFileHandler statsFileHandler;
     private final BucketMode bucketMode;
+    @Nullable private long commitTimeout;
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:137]: remove @Nullable

File: paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java [hunk 8]
```diff
@@ -166,7 +167,8 @@ public FileStoreCommitImpl(
             BucketMode bucketMode,
             @Nullable Integer manifestReadParallelism,
             List<CommitCallback> commitCallbacks,
-            int commitMaxRetries) {
+            int commitMaxRetries,
+            @Nullable long commitTimeout) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java:171]: @Nullable

Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java]: Done. Thanks for reminder @JingsongLi

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**